src/firefly/java/edu/caltech/ipac/firefly/server/ws/WebDAVWorkspaceHandler.java

```diff
@@ -31,10 +31,13 @@ public WorkspaceManager withCredentials(WsCredentials cred) {
 
         synchronized (cred) {
             String id = cred.getWsId();
```

Comment:
cred did not override equals.  I doubt if this synchronized does anything.  Plus, I don't think a wsPool is necessary.  The manager is meant as a way to encapsulate logic.  It's should be fine for single use.

Reply:
I don't think it will work as expected either. However, I would like to know what you are trying to accomplish.  What will happen it is is not synchronized. It it is necessary there might be another way to do it.